src/ContainerAwareTrait.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Dhii\Data\Container;
+
+use Exception as RootException;
+use InvalidArgumentException;
+
+/**
+ * Functionality for retrieval of a container.
+ *
+ * @since [*next-version*]
+ */
+trait ContainerAwareTrait
+{
+    /**
+     * The container.
+     *
+     * @since [*next-version*]
+     *
+     * @var ContainerInterface|null
+     */
+    protected $container;
+
+    /**
+     * Retrieves the container associated with this instance.
+     *
+     * @since [*next-version*]
+     *
+     * @return ContainerInterface|null The container, if any.
+     */
+    protected function _getContainer()
+    {
+        return $this->container;
+    }
+
+    /**
+     * Associates a container with this instance.
+     *
+     * @since [*next-version*]
+     *
+     * @param ContainerInterface|null $container The container.
+     *
+     * @throws InvalidArgumentException If not a valid container.
+     */
```

Comment:
Missing `return` tag

Reply:
Here, absence of the `@return` tag indicates that the returned value is unknown, and should not be relied on.

Reply:
The return type is known: it is the instance the method is called on. FYI: Setters in traits _can_ have `@return $this`.

Reply:
I agree. In fact, I think that chaining is not something we should be doing at all. I do, however, leave the opportunity for other implementations to implement chaining, which is why the type is undocumented, instead of documenting `null` or `void`.